src/viewProviders/schemas.test.ts

```diff
@@ -255,154 +220,298 @@ describe("SchemasViewProvider search behavior", () => {
+      // Need to pause an iota to get the refresh to be called, is after first await in the block.
+      await clock.tickAsync(100);
+
+      assert.ok(resetFake.notCalled);
+      assert.ok(updateTreeViewDescriptionFake.calledOnce);
+      assert.ok(refreshFake.calledOnce);
+    });
+
+    for (const currentRegistry of [TEST_LOCAL_SCHEMA_REGISTRY, null]) {
+      it(`Firing environmentChanged when SR set a ${currentRegistry?.environmentId} environment SR and event is for other env should do nothing`, () => {
+        const resetFake = sandbox.fake();
+        const updateTreeViewDescriptionFake = sandbox.fake();
+        const refreshFake = sandbox.fake();
+
+        sandbox.replace(provider, "reset", resetFake);
+        sandbox.replace(provider, "updateTreeViewDescription", updateTreeViewDescriptionFake);
+        sandbox.replace(provider, "refresh", refreshFake);
+
+        // Be set to a SR NOT within the environment being updated, or null.
+        provider.schemaRegistry = currentRegistry;
+
+        // Call the event handler against some other environment.
+        provider.environmentChangedHandler({
+          id: TEST_CCLOUD_ENVIRONMENT_ID,
+          wasDeleted: false,
+        });
+
+        // Should not have called any of these
+        assert.ok(resetFake.notCalled);
+        assert.ok(updateTreeViewDescriptionFake.notCalled);
+        assert.ok(refreshFake.notCalled);
+      });
+    }
   });
 
-  it("Viewing same schema registry, when schema added, should add to map + call refresh()", () => {
-    // set to be viewing a schema registry
-    provider.schemaRegistry = TEST_CCLOUD_SCHEMA_REGISTRY;
+  describe("ccloudConnectedHandler", () => {
+    for (const connected of [true, false]) {
+      it(`does nothing if no registry set, connected: ${connected}`, async () => {
+        provider.schemaRegistry = null;
 
-    // and this subject is in the map
-    subjectsInTreeView.set(TEST_CCLOUD_SUBJECT.name, TEST_CCLOUD_SUBJECT);
+        // Call the handler with true or false, should not matter.
+        await provider.ccloudConnectedHandler(connected);
 
-    const event: SubjectChangeEvent = {
-      subject: TEST_CCLOUD_SUBJECT_WITH_SCHEMAS,
-      change: "added",
-    };
-    schemaSubjectChanged.fire(event);
+        // Should not have called .reset()
+        assert.ok(resetStub.notCalled);
+      });
 
-    // Should have added to the map
-    assert.strictEqual(subjectsInTreeView.size, 1);
-    assert.deepStrictEqual(
-      subjectsInTreeView.get(TEST_CCLOUD_SUBJECT.name),
-      TEST_CCLOUD_SUBJECT_WITH_SCHEMAS,
-    );
+      it(`Should not reset if not viewing a CCloud schema registry, connected: ${connected}`, async () => {
+        provider.schemaRegistry = TEST_LOCAL_SCHEMA_REGISTRY;
 
-    // Should have called .refresh()
-    assert.ok(refreshStub.calledOnce);
-  });
-});
+        // Call the handler with true or false, should not matter.
+        await provider.ccloudConnectedHandler(true);
```

Comment:
Should this use the `connected` boolean and/or change the assertion?
```suggestion
        await provider.ccloudConnectedHandler(connected);
```